fd.ops.reshape seems to get the wrong input shape.

Reproducer:

```
import torch
from nvfuser import FusionDefinition

def test_expand_reshape():
    def expand_reshape() -> FusionDefinition:
        with FusionDefinition() as fd:
            x = fd.define_tensor([4, 5], [True, True])
            y = fd.ops.broadcast_in_dim(x, [4, 5, 6], [0, 1])
            y = fd.ops.reshape(y, [40, 3])
            fd.add_output(y)
        return fd

    fd = expand_reshape()

    x = torch.rand(4, 5, device="cuda")
    y = fd.execute([x])[0]
    print(y.shape)
```

```
$ pytest test.py -s
============================================================================================= test session starts =============================================================================================
platform linux -- Python 3.10.11, pytest-7.4.2, pluggy-1.3.0
rootdir: /opt/pytorch/nvfuser
plugins: xdist-3.3.1, hypothesis-6.88.1
collected 1 item

test.py F

================================================================================================== FAILURES ===================================================================================================
_____________________________________________________________________________________________ test_expand_reshape _____________________________________________________________________________________________

    def test_expand_reshape():
        def expand_reshape() -> FusionDefinition:
            with FusionDefinition() as fd:
                x = fd.define_tensor([4, 5], [True, True])
                y = fd.ops.broadcast_in_dim(x, [4, 5, 6], [0, 1])
                y = fd.ops.reshape(y, [40, 3])
                fd.add_output(y)
            return fd

>       fd = expand_reshape()

test.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test.py:6: in expand_reshape
    with FusionDefinition() as fd:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self =
def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[4, 5], contiguity=[True, Tru...  V8 = fd.define_vector([S6, S7], dtype=DataType.Int)
    T9 = fd.ops.reshape(T5, new_shape=V8)
    fd.add_output(T9)


type = None, value = None, traceback = None

    def __exit__(self, type, value, traceback):
>       self._finalize_definition()
E       RuntimeError: original_num_elements == new_num_elements INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/transform_view.cpp":372, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Total element counts across view operation must match: 20 vs 120
E       Exception raised from AnalyzeViewTransformation at /opt/pytorch/nvfuser/csrc/transform_view.cpp:372 (most recent call first):
E       frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x8d (0x7f888d9e37c7 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #1: nvfuser::nvfErrorFail(char const*, char const*, unsigned int, char const*, std::string const&) + 0x53 (0x7f888db25f43 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #2: nvfuser::analyzeView(nvfuser::TensorView const*, std::vector<long, std::allocator<long> > const&, std::vector<long, std::allocator<long> > const&) + 0x10a2 (0x7f888df68392 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #3: nvfuser::reshape(nvfuser::TensorView*, std::vector<long, std::allocator<long> > const&, std::vector<long, std::allocator<long> > const&) + 0xab (0x7f888ddc931b in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #4: nvfuser::reshape(nvfuser::TensorView*, std::vector<nvfuser::Val*, std::allocator<nvfuser::Val*> > const&) + 0x3f0 (0x7f888ddcae30 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #5: nvfuser::python_frontend::ReshapeOpRecord::operator()(nvfuser::python_frontend::FusionState&) + 0x4e (0x7f88800b30fe in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #6: nvfuser::python_frontend::FusionState::buildFusionIr(nvfuser::Fusion*) + 0xac (0x7f888df9e55c in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #7: nvfuser::python_frontend::FusionDefinition::finalizeDefinition() + 0x114 (0x7f888df99f04 in /usr/local/lib/python3.10/site-packages/torch/lib/libnvfuser_codegen.so)
E       frame #8: <unknown function> + 0xdd237 (0x7f888005d237 in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #9: <unknown function> + 0x16d927 (0x7f88800ed927 in /opt/pytorch/nvfuser/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
E       frame #10: <unknown function> + 0x14fcbb (0x56011c719cbb in /usr/local/bin/python3.10)
E       frame #11: _PyObject_MakeTpCall + 0x2ab (0x56011c7136ab in /usr/local/bin/python3.10)
E       frame #12: <unknown function> + 0x15b60f (0x56011c72560f in /usr/local/bin/python3.10)
E       frame #13: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #14: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #15: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #16: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #17: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #18: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #19: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #20: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #21: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #22: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #23: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #24: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #25: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #26: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #27: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)
E       frame #28: _PyObject_Call_Prepend + 0x6c (0x56011c722e0c in /usr/local/bin/python3.10)
E       frame #29: <unknown function> + 0x2282fd (0x56011c7f22fd in /usr/local/bin/python3.10)
E       frame #30: _PyObject_MakeTpCall + 0x2ab (0x56011c7136ab in /usr/local/bin/python3.10)
E       frame #31: _PyEval_EvalFrameDefault + 0x5d55 (0x56011c70ee35 in /usr/local/bin/python3.10)
E       frame #32: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #33: _PyEval_EvalFrameDefault + 0x6fe (0x56011c7097de in /usr/local/bin/python3.10)
E       frame #34: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #35: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #36: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #37: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #38: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #39: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #40: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #41: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)
E       frame #42: _PyObject_Call_Prepend + 0x6c (0x56011c722e0c in /usr/local/bin/python3.10)
E       frame #43: <unknown function> + 0x2282fd (0x56011c7f22fd in /usr/local/bin/python3.10)
E       frame #44: PyObject_Call + 0xe2 (0x56011c725e92 in /usr/local/bin/python3.10)
E       frame #45: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #46: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #47: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #48: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #49: _PyEval_EvalFrameDefault + 0x149c (0x56011c70a57c in /usr/local/bin/python3.10)
E       frame #50: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #51: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #52: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #53: _PyEval_EvalFrameDefault + 0x345 (0x56011c709425 in /usr/local/bin/python3.10)
E       frame #54: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #55: _PyEval_EvalFrameDefault + 0x149c (0x56011c70a57c in /usr/local/bin/python3.10)
E       frame #56: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #57: _PyEval_EvalFrameDefault + 0x2e8c (0x56011c70bf6c in /usr/local/bin/python3.10)
E       frame #58: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #59: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #60: <unknown function> + 0x15b315 (0x56011c725315 in /usr/local/bin/python3.10)
E       frame #61: _PyEval_EvalFrameDefault + 0x5165 (0x56011c70e245 in /usr/local/bin/python3.10)
E       frame #62: _PyFunction_Vectorcall + 0x73 (0x56011c71a123 in /usr/local/bin/python3.10)
E       frame #63: _PyObject_FastCallDictTstate + 0x18d (0x56011c712a0d in /usr/local/bin/python3.10)

nvfuser/__init__.py:41: RuntimeError
=========================================================================================== short test summary info ===========================================================================================
FAILED test.py::test_expand_reshape - RuntimeError: original_num_elements == new_num_elements INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/transform_view.cpp":372, please report a bug with repro script to NVFuser at https://github.c...
============================================================================================== 1 failed in 1.27s ==============================================================================================
```

For some reason, nvFuser thinks `original_num_elements` is 20, which is wrong. I suspect it's a collateral damage by #1043 .

--- a/csrc/ops/alias.cpp
+++ b/csrc/ops/alias.cpp
@@ -77,21 +77,21 @@ TensorView* tryStaticReshape(
     const std::vector<Val*>& new_sizes) {
   std::vector<int64_t> inp_sizes(inp_dom.size());
   for (const auto i : c10::irange(inp_dom.size())) {
-    auto id = inp_dom.at(i);
-    auto id_size = id->extent()->getInt();
-    if (!id_size.has_value()) {
+    IterDomain* id = inp_dom[i];
+    Val* id_size = id->getMaybeExpandedExtent();
+    if (!id_size->isConstInt()) {
       return nullptr;
     }
-    inp_sizes.at(i) = id_size.value();
+    inp_sizes[i] = id_size->evaluateInt();
   }
 
   std::vector<int64_t> out_sizes(new_sizes.size());
   for (const auto i : c10::irange(new_sizes.size())) {
-    auto id_size = new_sizes.at(i)->getInt();
-    if (!id_size.has_value()) {
+    Val* id_size = new_sizes[i];
+    if (!id_size->isConstInt()) {
       return nullptr;
     }
-    out_sizes.at(i) = id_size.value();
+    out_sizes[i] = id_size->evaluateInt();
   }
 
   // Both inputs are outputs are static. Just use the static version

--- a/csrc/python_frontend/fusion_record.h
+++ b/csrc/python_frontend/fusion_record.h
@@ -336,7 +336,7 @@ struct ReshapeOpRecord : RecordFunctor {
   }
 
   void operator()(FusionState& fd) final {
-    auto arg = fd.getFusionState(args_.at(0).index)->template as<TensorView>();
+    TensorView* arg = fd.getFusionState(args_.at(0).index)->as<TensorView>();
     const std::vector<Val*>& new_shape =
         fd.getFusionStateVector(args_.at(1).index);
     auto output = reshape(arg, new_shape);

--- a/test/test_gpu_view.cpp
+++ b/test/test_gpu_view.cpp
@@ -2443,4 +2443,39 @@ TEST_F(GpuViewTest, ReshapeOfReshape) {
   NVF_CHECK(ref.equal(cg_outputs.at(0)));
 }
 
+// A reproducer for #1116.
+TEST_F(GpuViewTest, ExpandedBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in = makeContigConcreteTensor({4, 5});
+  fusion.addInput(in);
+  TensorView* out = broadcast(in, {false, false, true});
+  out = expand(
+      out,
+      {IrBuilder::create<Val>(4),
+       IrBuilder::create<Val>(5),
+       IrBuilder::create<Val>(6)});
+  // tryStaticReshape failed to get the expanded extent, which is 6.
+  out = reshape(out, {IrBuilder::create<Val>(40), IrBuilder::create<Val>(3)});
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor =
+      at::randn({4, 5}, at::dtype(at::kFloat).device(at::kCUDA, 0));
+  at::Tensor expected_out_tensor =
+      in_tensor.unsqueeze(-1).expand({-1, -1, 6}).reshape({40, 3});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {in_tensor});
+  at::Tensor actual_out_tensor = fe.runFusion({in_tensor})[0];
+
+  testValidate(
+      &fusion,
+      {actual_out_tensor},
+      {in_tensor},
+      {expected_out_tensor},
+      __LINE__,
+      __FILE__);
+}
+
 } // namespace nvfuser

--- a/test/test_gpu_view.cpp
+++ b/test/test_gpu_view.cpp
@@ -50,7 +50,9 @@ namespace nvfuser {
 
 using namespace at::indexing;
 
-TEST_F(NVFuserTest, FusionViewDtypeSameSizeOutput_CUDA) {
+using GpuViewTest = NVFuserTest;
+
+TEST_F(GpuViewTest, FusionViewDtypeSameSizeOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -82,7 +84,7 @@ TEST_F(NVFuserTest, FusionViewDtypeSameSizeOutput_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_x_view}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionViewDtypeFailMismatchSize_CUDA) {
+TEST_F(GpuViewTest, FusionViewDtypeFailMismatchSize) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -100,7 +102,7 @@ TEST_F(NVFuserTest, FusionViewDtypeFailMismatchSize_CUDA) {
   ASSERT_ANY_THROW(view(x_add_bias, DataType::Half));
 }
 
-TEST_F(NVFuserTest, FusionViewAsRealOutput_CUDA) {
+TEST_F(GpuViewTest, FusionViewAsRealOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -150,7 +152,7 @@ TEST_F(NVFuserTest, FusionViewAsRealOutput_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_out}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeRfactorExtentReplacement_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeRfactorExtentReplacement) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 
@@ -178,7 +180,7 @@ TEST_F(NVFuserTest, FusionReshapeRfactorExtentReplacement_CUDA) {
       executor_cache.fusion(), cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeOutput_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -212,7 +214,7 @@ TEST_F(NVFuserTest, FusionReshapeOutput_CUDA) {
       &fusion, outputs, aten_inputs, {at_x_reshape}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeFailMismatchSize_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeFailMismatchSize) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -233,7 +235,7 @@ TEST_F(NVFuserTest, FusionReshapeFailMismatchSize_CUDA) {
   ASSERT_ANY_THROW(reshape(x_add_bias, input_shape, output_shape));
 }
 
-TEST_F(NVFuserTest, FusionReshapeFailMulitDimInference_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeFailMulitDimInference) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -367,7 +369,7 @@ std::vector<reshape_example> all_reshape_examples = {
     {{2, 3, 2 * 2, 5}, {1, 6, 1, 2, 2, 5, 1}},
 };
 
-TEST_F(NVFuserTest, FusionReshapeReductionShmoo_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeReductionShmoo) {
   for (auto e : all_reshape_examples) {
     // Shmoo tests can occupy a lot of memory due to allocating many
     // different tensor sizes. So in order to avoid an OOM during this
@@ -453,7 +455,7 @@ void persistentViewAddFusion(
   }
 }
 
-TEST_F(NVFuserTest, FusionReshapePersistentShmoo_CUDA) {
+TEST_F(GpuViewTest, FusionReshapePersistentShmoo) {
   for (auto e : all_reshape_examples) {
     // Shmoo tests can occupy a lot of memory due to allocating many
     // different tensor sizes. So in order to avoid an OOM during this
@@ -511,25 +513,25 @@ void addViewGeluFusion(
   }
 }
 
-TEST_F(NVFuserTest, FusionReshapeSplit_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeSplit) {
   std::vector<int64_t> input_shape{80};
   std::vector<int64_t> output_shape{2, 4, 10};
   addViewGeluFusion(input_shape, output_shape);
 }
 
-TEST_F(NVFuserTest, FusionReshapeBroadcast_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeBroadcast) {
   std::vector<int64_t> input_shape{80};
   std::vector<int64_t> output_shape{1, 80};
   addViewGeluFusion(input_shape, output_shape);
 }
 
-TEST_F(NVFuserTest, FusionReshapeMerge_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMerge) {
   std::vector<int64_t> input_shape{2, 40, 7};
   std::vector<int64_t> output_shape{560};
   addViewGeluFusion(input_shape, output_shape);
 }
 
-TEST_F(NVFuserTest, FusionReshapeAllShmoo_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeAllShmoo) {
   for (auto e : all_reshape_examples) {
     // Shmoo tests can occupy a lot of memory due to allocating many
     // different tensor sizes. So in order to avoid an OOM during this
@@ -585,7 +587,7 @@ void geluViewAddFusion(
   }
 }
 
-TEST_F(NVFuserTest, FusionReshapeStride_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeStride) {
   for (const auto& e : all_reshape_examples) {
     geluViewAddFusion(e.first, e.second);
   }
@@ -634,12 +636,12 @@ void geluViewBinaryAddFusion(
   }
 }
 
-TEST_F(NVFuserTest, FusionReshapeBinary_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeBinary) {
   geluViewBinaryAddFusion({27454, 2}, {54908}, {7844, 7});
 }
 
 // Repro of issue #1493
-TEST_F(NVFuserTest, FusionReshapeConcreteDomain_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeConcreteDomain) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -672,7 +674,7 @@ TEST_F(NVFuserTest, FusionReshapeConcreteDomain_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeConcreteDomain2_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeConcreteDomain2) {
   constexpr int kAxis = -1;
   std::vector<int64_t> input_shape = {19, 12, 7, 99};
   std::vector<int64_t> output_shape = {19, 3, 2772};
@@ -707,7 +709,7 @@ TEST_F(NVFuserTest, FusionReshapeConcreteDomain2_CUDA) {
 }
 
 // Repro of issue #1608
-TEST_F(NVFuserTest, FusionReshapeConcreteDomain3_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeConcreteDomain3) {
   std::vector<int64_t> input_shape = {14, 12, 8, 100};
   std::vector<int64_t> bcast_shape = {14, 12, 8, 1};
   std::vector<int64_t> other_shape = {14, 100, 96};
@@ -747,7 +749,7 @@ TEST_F(NVFuserTest, FusionReshapeConcreteDomain3_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_output}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeConcreteDomain4_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeConcreteDomain4) {
   std::vector<int64_t> shape1 = {3, 4, 5};
   std::vector<int64_t> shape2 = {3 * 4 * 5};
 
@@ -789,7 +791,7 @@ TEST_F(NVFuserTest, FusionReshapeConcreteDomain4_CUDA) {
       concrete_id->toString());
 }
 
-TEST_F(NVFuserTest, FusionReshapeConcreteDomain5_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeConcreteDomain5) {
   const std::vector<int64_t> shape1 = {12};
   const std::vector<int64_t> shape2 = {4, 3};
   const std::vector<int64_t> shape3 = {12, 5};
@@ -862,7 +864,7 @@ TEST_F(NVFuserTest, FusionReshapeConcreteDomain5_CUDA) {
   }
 }
 
-TEST_F(NVFuserTest, FusionFlattenAfterUnsqueezeOutput_CUDA) {
+TEST_F(GpuViewTest, FusionFlattenAfterUnsqueezeOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -898,7 +900,7 @@ TEST_F(NVFuserTest, FusionFlattenAfterUnsqueezeOutput_CUDA) {
       &fusion, outputs, aten_inputs, {at_x_reshape}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionComputeAtRootDomainMapWithView_CUDA) {
+TEST_F(GpuViewTest, FusionComputeAtRootDomainMapWithView) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -935,7 +937,7 @@ TEST_F(NVFuserTest, FusionComputeAtRootDomainMapWithView_CUDA) {
       tv1->axis(1)->toString());
 }
 
-TEST_F(NVFuserTest, FusionExpandRepro_CUDA) {
+TEST_F(GpuViewTest, FusionExpandRepro) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -969,7 +971,7 @@ TEST_F(NVFuserTest, FusionExpandRepro_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {out}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionExpandView1_CUDA) {
+TEST_F(GpuViewTest, FusionExpandView1) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 
@@ -1002,7 +1004,7 @@ TEST_F(NVFuserTest, FusionExpandView1_CUDA) {
       executor_cache.fusion(), cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionExpandView2_CUDA) {
+TEST_F(GpuViewTest, FusionExpandView2) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 
@@ -1032,7 +1034,7 @@ TEST_F(NVFuserTest, FusionExpandView2_CUDA) {
       executor_cache.fusion(), cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeTransformCache_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeTransformCache) {
   auto assert_matches = [](reshape_example example_0,
                            reshape_example example_1) {
     NVF_ERROR(
@@ -1180,7 +1182,7 @@ TEST_F(NVFuserTest, FusionReshapeTransformCache_CUDA) {
       {{19, 3 * 4, 7, 99}, {19, -1, 3}}, {{19, 3 * 5, 7, 99}, {19, -1, 3}});
 }
 
-TEST_F(NVFuserTest, FusionReshapeIdGraph_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeIdGraph) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -1237,7 +1239,7 @@ TEST_F(NVFuserTest, FusionReshapeIdGraph_CUDA) {
       tv2->getRootDomain()[3], tv12->getRootDomain()[3]));
 }
 
-TEST_F(NVFuserTest, FusionReshapeVectorize_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeVectorize) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -1282,7 +1284,7 @@ TEST_F(NVFuserTest, FusionReshapeVectorize_CUDA) {
   testValidate(&fusion, outputs, {input}, {tv_ref, tv_ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionExpandFlatten_CUDA) {
+TEST_F(GpuViewTest, FusionExpandFlatten) {
 #ifdef FBCODE_CAFFE2
   GTEST_SKIP() << "Fails accuracy on V100 32gb";
 #endif
@@ -1317,7 +1319,7 @@ TEST_F(NVFuserTest, FusionExpandFlatten_CUDA) {
       __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionIllegalReductionFlatten_CUDA) {
+TEST_F(GpuViewTest, FusionIllegalReductionFlatten) {
   EXPECT_THAT(
       []() {
         auto fusion = std::make_unique<Fusion>();
@@ -1334,7 +1336,7 @@ TEST_F(NVFuserTest, FusionIllegalReductionFlatten_CUDA) {
           testing::HasSubstr("Invalid end_dim")));
 }
 
-TEST_F(NVFuserTest, FusionReductionFlatten1_CUDA) {
+TEST_F(GpuViewTest, FusionReductionFlatten1) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 
@@ -1356,7 +1358,7 @@ TEST_F(NVFuserTest, FusionReductionFlatten1_CUDA) {
       executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionPwiseViewSchedule_CUDA) {
+TEST_F(GpuViewTest, FusionPwiseViewSchedule) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -1422,7 +1424,7 @@ TEST_F(NVFuserTest, FusionPwiseViewSchedule_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t3}, {t2, t4, t5}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionSumViewSchedule_CUDA) {
+TEST_F(GpuViewTest, FusionSumViewSchedule) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -1489,7 +1491,7 @@ TEST_F(NVFuserTest, FusionSumViewSchedule_CUDA) {
 }
 
 // Make sure matching reshapes are segmented into the same kernel
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule1_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule1) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1532,7 +1534,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule1_CUDA) {
 }
 
 // Make sure reshapes of reshapes are correct
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule2_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule2) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1568,7 +1570,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule2_CUDA) {
 // Make sure broadcasts not on the reshape path that don't interfere with
 // reshape are segmented in one kernel and correctly trigger 2D pointwise
 // scheduling
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule3_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule3) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1635,7 +1637,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule3_CUDA) {
 
 // Make sure broadcasts through reshapes when not conflicting with reshape are
 // segmented into one kernel and trigger 2D pointwise scheduler.
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule4_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule4) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1695,7 +1697,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule4_CUDA) {
 
 // Make sure different reshapes that are consumed by the reference are segmented
 // into a single kernel.
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule5_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule5) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1738,7 +1740,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule5_CUDA) {
 }
 
 // Test reshape/transpose and its impact on vectorization
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule6_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule6) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1787,7 +1789,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule6_CUDA) {
 }
 
 // View with 3D reduction scheduling
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule7_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule7) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1831,7 +1833,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule7_CUDA) {
 }
 
 // View with 3D normalization scheduling
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule8_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule8) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1881,7 +1883,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule8_CUDA) {
 }
 
 // AlbertForMaskedLM repro https://github.com/csarofeen/pytorch/issues/2066
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule9_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule9) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1956,7 +1958,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule9_CUDA) {
 }
 
 // Simpler version of FusionReshapeMagicSchedule9_CUDA
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule10_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule10) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -1988,7 +1990,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule10_CUDA) {
 }
 
 // CamemBert repro
-TEST_F(NVFuserTest, FusionReshapeMagicSchedule11_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMagicSchedule11) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -2015,7 +2017,7 @@ TEST_F(NVFuserTest, FusionReshapeMagicSchedule11_CUDA) {
 
 // Make sure different reshapes that are consumed by the reference are segmented
 // into a single kernel.
-TEST_F(NVFuserTest, FusionReshapeMapping_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeMapping) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -2069,7 +2071,7 @@ TEST_F(NVFuserTest, FusionReshapeMapping_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t3}, {t6}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionLowerDivisibleSplits_CUDA) {
+TEST_F(GpuViewTest, FusionLowerDivisibleSplits) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -2143,7 +2145,7 @@ TEST_F(NVFuserTest, FusionLowerDivisibleSplits_CUDA) {
   }
 }
 
-TEST_F(NVFuserTest, FusionIssue2076_CUDA) {
+TEST_F(GpuViewTest, FusionIssue2076) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -2250,7 +2252,7 @@ TEST_F(NVFuserTest, FusionIssue2076_CUDA) {
 }
 
 // Simplify first to reproduce compute at issue
-TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
+TEST_F(GpuViewTest, FusionIssue2076_v2) {
   auto fusion_ptr = std::make_unique<Fusion>();
   Fusion& fusion = *fusion_ptr.get();
   FusionGuard fg(&fusion);
@@ -2299,7 +2301,7 @@ TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t1, t2}, {t4, t6}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeZeroDimInput_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeZeroDimInput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -2334,7 +2336,7 @@ TEST_F(NVFuserTest, FusionReshapeZeroDimInput_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_prod}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeZeroDimOutput_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeZeroDimOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -2375,7 +2377,7 @@ TEST_F(NVFuserTest, FusionReshapeZeroDimOutput_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_prod}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, FusionReshapeZeroDimInputOutput_CUDA) {
+TEST_F(GpuViewTest, FusionReshapeZeroDimInputOutput) {
   Fusion fusion;
   FusionGuard fg(&fusion);
 
@@ -2412,7 +2414,7 @@ TEST_F(NVFuserTest, FusionReshapeZeroDimInputOutput_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {at_prod}, __LINE__, __FILE__);
 }
 
-TEST_F(NVFuserTest, ReshapeOfReshape_CUDA) {
+TEST_F(GpuViewTest, ReshapeOfReshape) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
 

